js/mail.js

```diff
@@ -47,14 +47,22 @@ function render(config, template) {
 }
 
 function loadTime(config, template) {
-  $.ajax(config.timeServer, {
+  var extraParams = '';
+  if (overrideActiveForAutomatedTests()) {
+    // setting this parameter switches the backend to demo mode
+    extraParams = '?currentTime=2099-12-24T18:00:00';
+    config.mail.recipient = "oriyungjkthx@mailinator.com";
+  }
```

Comment:
I don't know what these parameters trigger in the backend, but we need to be aware that users can find this in the source code, if we keep it in production. And I am pretty sure that someone will then try to manually do a request with those parameters.

If we ensure that the production backend ignores these parameters and there is no way for a user to get into this backend demo mode in production, we can keep it in.

Reply:
Not sure about removing the demo mode parameter in the production backend. Has its pros, but we also lose the ability to run the test suite against production (which we otherwise could do no problem).

Even if we leave it in, demo mode causes the backend to always return "TestSecret" as the secret. If anyone mails that to the hotel, well ... it's the same as sending a mail with any other wrongly guessed secret, and it's rather obviously wrong at that.

I don't particularly like having testing code in the frontend, but I really do want to be able to run automated tests for both situations (countdown still running and countdown finished). I'm definitely open for better ideas on how to get this parameter in, though. I considered having the ui test pass in the complete extraParams, but that doesn't help as its code is also public.

Reply:
If we want to run the test suite against production, then, by definition, the testing code also has to be in production. Nothing super wrong about it, just wanted to point out that this is something users can see 🙂 

As we are aware of that now, I'll merge this PR